### PR TITLE
Add minimal web server

### LINKED ISSUE #1: Move multidimensional scaling code into a minimal web server
Move the multidimensional scaling code in [Andromeda_IMG.ipynb](https://github.com/Imageomics/Andromeda/blob/main/Andromeda_IMG.ipynb) into a minimal web server.
This is a first step to deploying this project as a stand alone website.

### PR DESCRIPTION
**Pull request by johnbradley**
Adds flask web server that allows users to upload a CSV and perform dimensional reduction and inverse dimensional reduction.  Also adds tests for pre-existing dimension reduction code and automates running these tests. The dimension reduction tests are still rough and should be improved with better data and checks.

Fixes #1

### CODE REVIEW

**Comment by thompsonmj** [comment]
We might also move the `test_*.py` files into a `tests/` directory and add an `__init__.py` file there.

**Comment by egrace479** [comment]
I agree with Matt's suggestion on moving the test files. Also, there's a typo at line 57 of the readme: should be 'x' is the x coordinate, instead of 'y' for both.

**Comment by johnbradley** [comment]
@thompsonmj @egrace479 I moved the tests to their own directory and updated the README.

**Comment by thompsonmj** [comment]
Since it's going to a Linux server, I guess we don't _really_ need generalized filesystem paths!

**Comment by egrace479** [comment]
LGTM

**Review by egrace479 — APPROVED** [review]
LGTM

**Review by thompsonmj — APPROVED** [review]

File: andromeda/test_dataset.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        dataset = ds.create_dataset(user_file)
+
+        self.assertEqual(dataset.id, "9b4973f4-eba8-41b8-a3f9-acbadf49a2ca")
+        user_file.save.assert_called_with('/tmp/9b4973f4-eba8-41b8-a3f9-acbadf49a2ca.csv')
+
+        # Test that we read CSV where expected
+        df = dataset.read_dataframe()
+
+        self.assertEqual(df, mock_pd.read_csv.return_value)
+        mock_pd.read_csv.assert_called_with('/tmp/9b4973f4-eba8-41b8-a3f9-acbadf49a2ca.csv')
```
[the remark this answers is not in the document]
Comment (thompsonmj) [reply andromeda/test_dataset.py]: It passed ✔
Maybe adding `windows-latest` and `macos-latest` to the Actions config would be good too?
```
jobs:
  build:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
        python-version: ["3.11"]
# ...
```
Is this a best practice that should be used for all tests? It's only meant to run on a server, but testing locally during development could be convenient, and having cross-platform CI could make it even smoother.

[the remark this answers is not in the document]
Comment (johnbradley) [reply andromeda/test_dataset.py]: There are some limits on the number of action minutes GitHub provides for free. Also macos-latest uses up 10x the minutes. I'll create an issue to discuss this further.

**Pull request merged**